internal/reporter/reporter.go

```diff
@@ -45,24 +50,32 @@ func (n *noLint) Matches(position token.Position) bool {
 	return n.filename == position.Filename && (n.line == position.Line || n.line+1 == position.Line)
 }
 
-// Pass is a wrapper around *analysis.Pass that accounts for nolint directives. Please never
-// initialize this type directly, only through PassWithNoLint. If you initialize this type
+// Pass is a wrapper around *analysis.Pass that accounts for nolint directives as well as any
+// other functionality that it is configured with during initialization with the factory function.
+// Please never initialize this type directly, only through NewPass. If you initialize this type
 // directly, it will essentially be a no-op wrapper around *analysis.Pass.
 type Pass struct {
 	noLints []noLint
 	linter  string
 
 	*analysis.Pass
+
+	// Functional option supplied configuration
```

Comment:
Describe what this controls:
```suggestion
	// True if this should treat linter issues as warnings.
```